src/lib/predicates/isSome.ts

```diff
@@ -1,7 +1,7 @@
 import { Predicate, Some } from '../types'
 
 /**
- * Checks if a value is a value other then `null` or `undefined`.
+ * Checks if a value is a value other then `null`, `undefined`, or not NaN.
  */
 export const isSome: Predicate<Some> = (value: unknown): value is Some =>
-  value != null && value === value
+  value != null && Number.isNaN(value) === false
```

Comment:
A comment explaining that `value === value` is to check against `NaN` check would be preferred over the code change because this is a lot more expensive. `isSome` is quite a 'low level' predicate which is used a lot (by/in other predicates).